### PR TITLE
Bump rest.js versions to 2.17.0 for new sharing capabilities

### PR DESCRIPTION
**Pull request by rweber-esri**
[144791](https://esriarlington.tpondemand.com/entity/144791-search-filter-and-select-groups-to)

Bump rest.js versions to 2.17.0 for new sharing capabilities.

### CODE REVIEW

**Review by dbouwman — APPROVED** [review]
This looks right to me... but human-yarn @tomwayson should double check the math

**Comment by rweber-esri** [comment]
@tomwayson 

Only things that depend on _new_ functionality (i.e. I added removeUsers to REST.js) should have their `peerDependencies` bumped, which nothing depends on _new_ functionality, so `peerDepencendies` left as-is. Looks like `common`, `initiatives`, `sites` & `surveys` use REST.js `shareItemWithGroup` and `unshareItemWithGroup`, so bumped their `devDependencies` to latest REST.js `2.17.0`.

File: packages/sites/src/drafts/save-published-status.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { IModel, IHubRequestOptions } from "@esri/hub-common";
 import { updateSite } from "../update-site";
 import { updatePage, isPage } from "../pages";
 import { isSite } from "../is-site";
+import { IUpdateItemResponse } from "@esri/arcgis-rest-portal";
```
Comment (tomwayson) [inline packages/sites/src/drafts/save-published-status.ts:5]: if this interface is new in 2.17, we need to bump `@esri/arcgis-rest-portal` in the sites' `peerDependencies` as well as the `devDependencies`

**Review by tomwayson — CHANGES_REQUESTED** [review]

File: packages/sites/src/drafts/save-published-status.ts (hunk 1 above)
Comment (tomwayson) [reply packages/sites/src/drafts/save-published-status.ts:5]: sorry, I didn't mean to block this - I meant to add that as a comment, not request for change.

And yes, you were on exactly the right track.

I'm going to approve this PR, it's good as is as long as `IUpdateItemResponse` was not added in 2.17

**Pull request merged**